### PR TITLE
FEATURE: Group write permission check

### PR DESCRIPTION
**Pull request by wexi79**
Added a method that creates a test folder into the Flow root path, creates a test file into the folder and checks that the created file has write permissions for group. Test file and folder are removed after the check, method returns true if group has write permission and false if not.

If group write fails, an error message with instructions is displayed.

### CODE REVIEW

**Comment by wexi79** [comment]
This could be extended to checking all the other permissions of the files created too, if needed.

**Comment by wexi79** [comment]
@kitsunet for some reason I can't request reviewers for this PR, but could you look at this?

**Comment by wexi79** [comment]
@kdambekalns what do you mean "remove alignments"? I had to format the code in that way to get it to pass the automatic styling check...

**Comment by kdambekalns** [comment]
Oh well. Then StyleCI is using default settings…

File: Classes/Core/BasicRequirements.php [hunk 1]
```diff
@@ -165,7 +163,29 @@ protected function checkFilePermissions()
...
 
-        return null;
+    /**
+     * Check that files are being created with group write permissions.
+     *
+     * @return bool
+     */
+    protected function checkFileGroupWritePermission()
+    {
+        $testfolder = FLOW_PATH_ROOT . 'Test';
```
Comment (kdambekalns) [inline Classes/Core/BasicRequirements.php]: Actually, is this required to be writable? On a "really secure system" I'd only make `Web/_Resources`, `Configuratio` (_sigh_) and `Data` writable, as that should suffice. So maybe test in `FLOW_PATH_DATA`?

**Comment by wexi79** [comment]
The Data folder need to be created (by the setfilepermissions script) under FLOW_PATH_ROOT so what we actually do is test that we can create a similar folder with the correct permissions.

**Comment by wexi79** [comment]
So basically we are testing that the cli user can access all files and folders in the Neos installation directory, is there something that the cli user should not be able to access in a setup where the cli user has no sudo rights but the initial installation is done by a sysadmin user?

File: Classes/Core/BasicRequirements.php (hunk 1 above)
Comment (kitsunet) [reply Classes/Core/BasicRequirements.php]: `Configuration` no longer needed YAY

**Comment by kitsunet** [comment]
This is not using the CLI user though but the web user. I would find this kind of check pretty great but agree that ideally for the Web user this should only happen for Data and Web

**Comment by mhsdesign** [comment]
superseded by rewrite in version 6

**Pull request closed without merging**